schema_generator/validation.py

```diff
@@ -271,8 +271,12 @@ def boolean_list_property(question):
     }}
 
 
-def price_string(optional):
-    pattern = r"^\d{1,15}(?:\.\d{1,5})?$"
+def price_string(optional, decimal_place_restriction=False):
+    pattern = r"^\d{1,15}(?:\.\d{1,5})?$"  # up to 5 decimal places allowed eg 90, 90.1, 90.12345
+    restricted_pattern = r"^[1-9](?:\d{1,15})?(?:\.\d{2})?$|^0\.(?!00)\d{2}$"  # 0dp or 2dp only allowed eg 90 or 90.12
```

Comment:
It would be nice if this restricted pattern were more forgiving about supplying decimals or not. It feels like we should still accept things like '90.' and '90.0', which still correlate to valid prices if we extrapolate appropriately. We want to enforce the new rules without making it less user-friendly.

Also we should keep the regexes semantically the same. We've added the `[1-9]` prefix at the start, so we should drop the repeat clause `{1,15}` down to `{1,14}` to keep us at the same overall max length.

Reply:
Good point on the `{1,14}`. Will implement.

Reply:
Comment from trello: After discussion with Sam and Tobi, we have decided to be more permissive with the input we are allowing.

If a user put 90. then we would allow this input and strip out the . before saving it. And if a user put 90.1, we would save it as 90.10.

The API validation will still remain as tight but the content loader will deal with the formatting before we send it.